Droidcon-Boston/app/src/main/java/com/mentalmachines/droidcon_boston/views/detail/AgendaDetailFragment.kt

```diff
@@ -0,0 +1,182 @@
+        tv_agenda_detail_time.text = itemData.startTime + " - " + itemData.endTime
+
+        fab_agenda_detail_bookmark.setOnClickListener({
+
+            val nextBookmarkStatus = !userAgendaRepo.isSessionBookmarked(scheduleDetail.id)
+            userAgendaRepo.bookmarkSession(scheduleDetail.id, nextBookmarkStatus)
+
+            Snackbar.make(agendaDetailView,
+                    if (nextBookmarkStatus)
+                        getString(R.string.saved_agenda_item)
+                    else getString(R.string.removed_agenda_item),
+                    Snackbar.LENGTH_SHORT).show()
+
+            showBookmarkStatus(scheduleDetail)
+        })
+
+        populateSpeakersInformation(itemData)
+    }
+
+    private fun fetchDataFromFirebase(itemData: ScheduleRow) {
+        firebaseHelper.speakerDatabase.orderByChild("name")
+                .addValueEventListener(object : ValueEventListener {
+                    override fun onDataChange(dataSnapshot: DataSnapshot) {
+                        for (speakerSnapshot in dataSnapshot.children) {
+                            val detail = speakerSnapshot.getValue(ScheduleEventDetail::class.java)
+                            if (detail != null) {
+                                scheduleDetail = detail.toScheduleDetail(itemData)
+                                showAgendaDetail(scheduleDetail)
+                            }
+
+                        }
+
+                    }
+
+                    override fun onCancelled(databaseError: DatabaseError) {
+                        Log.e(javaClass.canonicalName, "detailQuery:onCancelled", databaseError.toException())
+                    }
+                })
+    }
+
+
+    private fun populateSpeakersInformation(itemData: ScheduleRow) = when {
+        itemData.speakerNames?.size == 0 -> tv_agenda_detail_speaker_name.visibility = View.GONE
+        else -> {
+            var speakerNames = ""
+            var marginValue = 28
+            itemData.speakerNames?.forEach {
+                val orgName: String? = itemData.speakerNameToOrgName?.get(it)
+                // append org name to speaker name
+                speakerNames += it + when {
+                    orgName != null -> " - $orgName"
+                    else -> {
+                        // Do nothing
+                    }
+                }
+
+                if (itemData.speakerNames?.size!! > 1) {
+                    tv_agenda_detail_speaker_title.text = getString(string.str_speakers)
+
+                    // if the current speaker name is not the last then add a line break
+                    if (!it.equals(itemData.speakerNames?.last())) {
+                        speakerNames += "\n"
+                    }
+                } else {
+                    tv_agenda_detail_speaker_title.text = getString(string.str_speaker)
+                }
+
+
+                // Add an imageview to the relative layout
+                val tempImg = ImageView(activity)
```

Comment:
Couldn't do this in the layout?

Reply:
It is dynamic. Serves one speaker, two or more.